src/Uno.UI/UI/Xaml/Controls/Popup/Popup.cs

```diff
@@ -28,8 +31,82 @@ internal PopupPanel PopupPanel
 		}
 
 		public static readonly DependencyProperty PopupPanelProperty =
-			DependencyProperty.Register("PopupPanel", typeof(PopupPanel), typeof(Popup), new PropertyMetadata(null, (s, e) => ((Popup)s)?.OnPopupPanelChanged(e)));
+			DependencyProperty.Register("PopupPanel", typeof(PopupPanel), typeof(Popup), new PropertyMetadata(null, (s, e) => ((Popup)s)?.OnPopupPanelChanged((PopupPanel)e.OldValue, (PopupPanel)e.NewValue)));
 
-		partial void OnPopupPanelChanged(DependencyPropertyChangedEventArgs e);
+		private void OnPopupPanelChanged(PopupPanel oldHost, PopupPanel newHost)
+		{
+			// Note: On UWP if the popup is light dismissable, the popup is closed as soon as we get a PointerPressed
+			//		 Controls under the popup does not receive any Pointer event while the popup is open,
+			//		 and when light dismissed they receive the PointerReleased but not the pressed
+
+			if (oldHost != null)
+			{
+				oldHost.PointerPressed -= _handleIfOpenedAndTryDismiss; // Cf. comment about child pressed below
+
+				oldHost.PointerEntered -= _handleIfOpened;
+				oldHost.PointerExited -= _handleIfOpened;
+				oldHost.PointerMoved -= _handleIfOpened;
+				oldHost.PointerReleased -= _handleIfOpened;
+				oldHost.PointerCanceled -= _handleIfOpened;
+				oldHost.PointerCaptureLost -= _handleIfOpened;
+				oldHost.PointerWheelChanged -= _handleIfOpened;
+			}
+
+			if (newHost != null)
+			{
+				newHost.PointerPressed += _handleIfOpenedAndTryDismiss; // Cf. comment about child pressed below
```

Comment:
When do we remove the events ?

Reply:
They are removed if the `PopupPanel` changes, otherwise as the link between `PopupPanel` and the `Popup` is 1-1, that should not leak as everything will be collected at once.